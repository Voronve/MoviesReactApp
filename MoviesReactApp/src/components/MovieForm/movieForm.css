.movieForm {
    font: 15px 'Montserrat', sans-serif;
    color: white;
    background-color: #232323;
    display: flex;
    flex-wrap: wrap;
    max-width: 900px;
    margin: auto;
  }

  .confirmMessage {
    margin: 30px 5px;
    width: 100%;
    text-align: center;
  }

  .column-2-3 {
    box-sizing: border-box;
    width: 66.66%;
    padding: 10px;
  }

  .column-1-3 {
    box-sizing: border-box;
    width: 33.33%;
    padding: 10px;
  }

  .full-width {
    width: 100%;
    padding: 10px;
  }

  .movieForm label {
    color: red;
    display: block;
    margin-bottom: 5px;
  }

  .movieForm input,
  .movieForm textarea,
  .movieForm select {
    color: white;
    background-color: #424242;
    box-sizing: border-box;
    width: 100%;
    padding: 5px;
    font-size: 14px;
    margin-bottom: 15px;
  }

  .movieForm .checkboxes {
    display: flex;
    flex-direction: column;
  }

  .movieForm .checkbox {
    display: flex;
    align-items: center;
    margin-bottom: 5px;
  }

  .movieForm .checkbox input {
    margin-right: 3px;
  }

  .movieForm .buttonBlock {
    padding: 10px;
    display: flex;
    width: 100%;
    justify-content: end;
  }

  .movieForm button {
    padding: 10px 20px;
    margin-left: 10px;
    border-radius: 4px;
    cursor: pointer;
  }

  .movieForm .submitButton {
    background-color: #F65261;
    border: none;
    color: white;
  }

  .movieForm .resetButton {
    background-color: inherit;
    border: 2px solid red;
    border-radius: 4px;
    color: red;
  }