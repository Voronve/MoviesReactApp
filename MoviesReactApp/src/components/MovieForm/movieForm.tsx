import React, { FormEvent, KeyboardEvent, ChangeEvent, useState } from 'react';
import { MovieInfo } from '../MovieListPage/movieListPage';
import './movieForm.css';

export interface MovieFormProps {
    movieInfo?: MovieInfo,
    onSubmit: () => void;
}

/** Form for searching movies */
export function MovieForm({ movieInfo, onSubmit }: MovieFormProps) {

    const handleSubmit = (event: React.FormEvent) => {
        event.preventDefault();
        onSubmit();
    }

    const handleReset = (event: React.MouseEvent) => {
        event.preventDefault();
        alert('Call from reset');
    }

    return (
         <form className='movieForm'>
            <div className="column-2-3">
                <label htmlFor="title">Title:</label>
                <input type="text" id="title" name="title" defaultValue={movieInfo ? movieInfo.title : ''}/>

                <label htmlFor="movie-url">Movie URL:</label>
                <input type="text" id="movie-url" name="movie-url" defaultValue={movieInfo ? movieInfo.poster_path : ''}/>

                <label>Genre:</label>
                <input type="text" id="janres" name="janres" defaultValue={(movieInfo && movieInfo.genres) ? movieInfo.genres.join(', ') : ''}/>
            </div>

            <div className="column-1-3">
                <label htmlFor="release-date">Release Date:</label>
                <input type="date" id="release-date" name="release-date" defaultValue={movieInfo ? movieInfo.release_date : ''}/>

                <label htmlFor="vote_average">Rating:</label>
                <input type="text" id="vote_average" name="vote_average" defaultValue={movieInfo ? movieInfo.vote_average : ''}/>

                <label htmlFor="runtime">Runtime:</label>
                <input type="text" id="runtime" name="runtime" defaultValue={movieInfo ? movieInfo.runtime : ''}/>
            </div>

            <div className="full-width">
                <label htmlFor="overview">Overview:</label>
                <textarea id="overview" name="overview" rows={4} defaultValue={movieInfo ? movieInfo.overview : ''}></textarea>
            </div>
            <div className="buttonBlock">
                <button className="resetButton" onClick={handleReset} >RESET</button>
                <button className="submitButton" type='submit' onSubmit={handleSubmit}>SUBMIT</button>
            </div>
        </form>
    );
}