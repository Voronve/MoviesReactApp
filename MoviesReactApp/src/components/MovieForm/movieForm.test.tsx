import { MovieForm } from './movieForm';
import { fireEvent, render, screen } from '@testing-library/react';

describe("Testing MovieForm component", () => {
    const onClick = jest.fn();
    const movieDataMock = {
        id: "123",
        poster_path: "Comedy/Barbie.jpg",
        title: "Barbie",
        release_date: 2023,
        genres: ["COMEDY", "HORROR"],
        vote_average: 8.9,
        runtime: "2h 34min",
        overview: "Test overview"
    }

    test("Testing movie form rendering", () => {
        const { asFragment } = render(<MovieForm onSubmit={onClick} />);

        expect(asFragment()).toMatchSnapshot();
    });

    test("Testing that filling movie form value with default data is working", () => {
        render(<MovieForm movieInfo={movieDataMock} onSubmit={onClick} />);

        const input = screen.getByDisplayValue("COMEDY, HORROR");

        expect(input).toBeInTheDocument();
    });

    test("Testing that callback function is working as expected", () => {
        render(<MovieForm movieInfo={movieDataMock} onSubmit={onClick} />);

        fireEvent.submit(screen.getByText('SUBMIT'));

        expect(onClick).toHaveBeenCalled();
    });
});