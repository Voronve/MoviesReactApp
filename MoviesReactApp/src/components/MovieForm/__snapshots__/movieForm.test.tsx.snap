// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MovieForm component Testing movie form rendering 1`] = `
<DocumentFragment>
  <form
    class="movieForm"
  >
    <div
      class="column-2-3"
    >
      <label
        for="title"
      >
        Title:
      </label>
      <input
        id="title"
        name="title"
        type="text"
        value=""
      />
      <label
        for="movie-url"
      >
        Movie URL:
      </label>
      <input
        id="movie-url"
        name="movie-url"
        type="text"
        value=""
      />
      <label>
        Genre:
      </label>
      <input
        id="janres"
        name="janres"
        type="text"
        value=""
      />
    </div>
    <div
      class="column-1-3"
    >
      <label
        for="release-date"
      >
        Release Date:
      </label>
      <input
        id="release-date"
        name="release-date"
        type="date"
        value=""
      />
      <label
        for="rating"
      >
        Rating:
      </label>
      <input
        id="rating"
        name="rating"
        type="text"
        value=""
      />
      <label
        for="runtime"
      >
        Runtime:
      </label>
      <input
        id="runtime"
        name="runtime"
        type="text"
        value=""
      />
    </div>
    <div
      class="full-width"
    >
      <label
        for="overview"
      >
        Overview:
      </label>
      <textarea
        id="overview"
        name="overview"
        rows="4"
      />
    </div>
    <div
      class="buttonBlock"
    >
      <button
        class="resetButton"
      >
        RESET
      </button>
      <button
        class="submitButton"
        type="submit"
      >
        SUBMIT
      </button>
    </div>
  </form>
</DocumentFragment>
`;
